packages/babel-helper-create-class-features-plugin/src/fields.js

```diff
@@ -100,6 +100,34 @@ const privateNameVisitor = {
   },
 };
 
+const privateInVisitor = {
+  BinaryExpression(path) {
+    const { operator, left, right } = path.node;
+    if (operator !== "in") return;
+    if (!path.get("left").isPrivateName()) return;
+
+    const { privateNamesMap } = this;
+    const { name } = left.id;
+
+    if (this.loose) {
+      const { id } = privateNamesMap.get(name);
+      path.replaceWith(template.expression.ast`
+        Object.prototype.hasOwnProperty.call(${right}, ${id})
+      `);
+      return;
+    }
+
+    const { id, static: isStatic } = privateNamesMap.get(name);
+
+    if (isStatic) {
+      path.replaceWith(template.expression.ast`${right} === ${this.classRef}`);
+      return;
+    }
+
+    path.replaceWith(template.expression.ast`${id}.has(${right})`);
```

Comment:
just curious; do these field WeakMaps have `WeakMap.prototype.has` copied onto them? or can i intercept babel private fields, even in spec mode, by replacing `WeakMap.prototype.has`?

Reply:
We do not secure the weak
maps, so this will do a prototype reference. Note that even `get` and `set` calls go through the prototype, too. We can't ensure the order of our evaluation, so the `WeakMap` could have already been patched by the time we reach the class definition.

Reply:
Of course - you can't protect against first-run code, but to be spec compliant you'd have to at least protect against later code. `Object.defineProperties(id, Object.getOwnPropertyDescriptors(WeakMap.prototype))` at `id` creation would address that.

Reply:
Currently, all of our output assumes that builtins haven't been modified.

Reply:
oof, ok - it's totally fine to assume they haven't been modified at module evaluation time (you don't really have a choice) but it's *not* fine to assume they haven't been modified later. that should be fixed asap (but not in this PR, obviously)